src/lsm/lsm_work_unit.c

```diff
@@ -117,17 +117,13 @@ __wt_lsm_get_chunk_to_flush(WT_SESSION_IMPL *session,
 	} else
 		chunk = (evict_chunk != NULL) ? evict_chunk : flush_chunk;
 
-	if (chunk != NULL) {
```

Comment:
I think ```chunk``` can be NULL here. If we loop through and don't find any chunks to evict or flush, we come to:

```
chunk = (evict_chunk != NULL) ? evict_chunk : flush_chunk;
```

Both ```evict_chunk``` and ```flush_chunk``` can be NULL - so ```chunk``` will be NULL.

We could add a test for that case explicitly and exit early. It would need to unlock the tree so it'd be a goto err with no error set, which I don't love.